app/models/weam.rb

```diff
@@ -163,16 +155,38 @@ def vet_tec_flags_for_approved?
   # If school is a vet tec provider, applicable law code is VET TEC ONLY,
   # and be a non college degree indicator
   def approved?
-    return false if poo_status.blank? || applicable_law_code.blank?
+    return false if applicable_law_code.blank?
 
-    return false unless poo_status =~ Regexp.new('aprvd', 'i')
+    return false unless poo_status_valid?
 
-    # VET TEC ONLY
-    return vet_tec_flags_for_approved? if applicable_law_code =~ Regexp.new(ALCVT, 'i')
+    return vet_tec_approved? if vet_tec?
+
+    return false if invalid_law_code?
 
-    # Other
-    return false if applicable_law_code =~ Regexp.new("#{ALC1}|#{ALC2}", 'i')
     flags_for_approved?
   end
+
+  private
+
+  ALC1 = 'educational institution is not approved'
+  ALC2 = 'educational institution is approved for chapter 31 only'
+  ALCVT = 'vet tec only'
```

Comment:
Just a style nitpick, but can we move these back to the top of the file with the other constant(s)?
That's where you can generally expect to find constants, ala [this style guide](https://github.com/rubocop-hq/ruby-style-guide#consistent-classes).
Moving them down here does not make the constants private.
If you really do want to make the constants private, you can use [private_constant](http://www.chrisrolle.com/en/blog/private-constants-in-ruby)